code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/notification/builders/AepPushNotificationBuilder.kt

```diff
@@ -0,0 +1,971 @@
+            createPendingIntent(
+                context,
+                trackerActivityClass,
+                actionUri,
+                null,
+                tag,
+                stickyNotification
+            )
+        notificationBuilder.setContentIntent(pendingIntent)
+    }
+
+    /**
+     * Sets the delete action for the notification.
+     *
+     * @param context the application [Context]
+     * @param trackerActivityClass the [Class] of the activity to set in the created pending intent for tracking purposes
+     * @param builder the [NotificationCompat.Builder] to attach the delete action
+     * notification
+     */
+    internal fun setNotificationDeleteAction(
+        context: Context,
+        trackerActivityClass: Class<out Activity>?,
+        builder: NotificationCompat.Builder,
+    ) {
+        val deleteIntent = Intent(PushTemplateConstants.NotificationAction.DISMISSED)
+        trackerActivityClass?.let {
+            deleteIntent.setClass(context.applicationContext, trackerActivityClass)
+        }
+        deleteIntent.setFlags(Intent.FLAG_ACTIVITY_SINGLE_TOP)
+        val intent = PendingIntent.getActivity(
+            context,
+            Random().nextInt(),
+            deleteIntent,
+            PendingIntent.FLAG_UPDATE_CURRENT or PendingIntent.FLAG_IMMUTABLE
+        )
+        builder.setDeleteIntent(intent)
+    }
+
+    /**
+     * Creates a pending intent for a notification.
+     *
+     * @param context the application [Context]
+     * @param trackerActivityClass the [Class] of the activity to set in the created pending intent for tracking purposes
+     * notification
+     * @param actionUri the action uri
+     * @param actionID the action ID
+     * @param stickyNotification [Boolean] if false, remove the notification after it is interacted with
+     * @return the created [PendingIntent]
+     */
+    internal fun createPendingIntent(
+        context: Context,
+        trackerActivityClass: Class<out Activity>?,
+        actionUri: String?,
+        actionID: String?,
+        tag: String?,
+        stickyNotification: Boolean
+    ): PendingIntent? {
+        val intent = Intent(PushTemplateConstants.NotificationAction.BUTTON_CLICKED)
+        trackerActivityClass?.let {
+            intent.setClass(context.applicationContext, trackerActivityClass)
+        }
+        intent.setFlags(Intent.FLAG_ACTIVITY_SINGLE_TOP)
+        intent.putExtra(PushTemplateConstants.PushPayloadKeys.TAG, tag)
+        intent.putExtra(PushTemplateConstants.PushPayloadKeys.STICKY, stickyNotification)
+        addActionDetailsToIntent(
+            intent,
+            actionUri,
+            actionID
+        )
+
+        // adding tracking details
+        return TaskStackBuilder.create(context)
+            .addNextIntentWithParentStack(intent)
+            .getPendingIntent(
+                Random().nextInt(),
+                PendingIntent.FLAG_UPDATE_CURRENT or PendingIntent.FLAG_IMMUTABLE
+            )
```

Comment:
TaskStackBuilder usage resulted in a bug recently in Messaging which was fixed. Can you apply the same fix as here please - https://github.com/adobe/aepsdk-messaging-android/pull/242/files